core/Tracker/GoalManager.php

```diff
@@ -677,6 +678,12 @@ protected function recordStandardGoals(VisitProperties $visitProperties, Request
             $conversionDimensions = ConversionDimension::getAllDimensions();
             $conversion = $this->triggerHookOnDimensions($request, $conversionDimensions, 'onGoalConversion', $visitor, $action, $conversion);
 
+            if ($this->isEventMatchingGoal($convertedGoal)
+                && !empty($convertedGoal['event_value_as_revenue'])
+            ) {
+                $conversion['revenue'] = ActionEvent::getEventValue($request);
```

Comment:
Don't know if it's any important or makes a difference... usually for the revenue column we do a 
```php
$revenue = Common::getRequestVar('revenue', $defaultGoalRevenue, 'float', $this->params);
$revenue = BaseConversion::roundRevenueIfNeeded($revenue)
```

I presume this logic cannot be in a dimension itself and be defined through the "onGoalConversion"? I can see various problems like "revenue  might be overwritten by other dimension", "it might not know if the goal is an event matching a goal etc", "it doesn't know that flag "event_value_as_revenue" is enabled, ... so all good if it is not possible.

I wonder though if it still worked if the conversion revenue is defined before line 679 before we trigger the hook, so dimension can still change it if wanted/needed. But this may not be something we really want and we may want to enforce that revenue in this case so all good probably.

Reply:
I thought about putting it in a dimension, but was unsure as to which dimension to put it in. Eg, does it go in EventValue and access goal data there? Or in a goal dimension?

I could maybe put it in an event handler in Goals if we make sure it overrides other changes. Or maybe a RequestProcessor.

Reply:
Yeah not sure either. Would need to be a dimension I suppose but not sure if the dimension would all the information it needs? I think it's all good like this just in case we don't know which order things are executed or so. Was just a random thought I wanted to mention :)